### PR TITLE
change treeItem icon

### PR DESCRIPTION
**Pull request by Eskibear**

### CODE REVIEW

File: src/Utils.ts [hunk 1]
```diff
@@ -1,29 +1,65 @@
...
+let EXTENSION_NAME: string;
+let EXTENSION_VERSION: string;
+let EXTENSION_AI_KEY: string;
 
 export namespace Utils {
-    export async function getProject(absolutePath: string, workspacePath: string, iconPath?: string): Promise<ProjectItem> {
-        if (await fs.pathExists(absolutePath)) {
-            const xml: string = await fs.readFile(absolutePath, "utf8");
+    export function loadPackageInfo(context: ExtensionContext): void {
+        const { publisher, name, version, aiKey } = fse.readJSONSync(context.asAbsolutePath("./package.json"));
```
Comment (jdneo) [inline src/Utils.ts]: await readJson()?

File: src/Utils.ts [hunk 2]
```diff
@@ -1,29 +1,65 @@
...
 import { Archetype } from "./model/Archetype";
 import { ProjectItem } from "./model/ProjectItem";
 import { IArchetype, IArchetypeCatalogRoot, IArchetypes, IPomRoot } from "./model/XmlSchema";
-const EXTENSION_NAME: string = "vscode-maven";
-const EXTENSION_ID: string = "eskibear.vscode-maven";
+
+let EXTENSION_PUBLISHER: string;
+let EXTENSION_NAME: string;
+let EXTENSION_VERSION: string;
+let EXTENSION_AI_KEY: string;
```
Comment (jdneo) [inline src/Utils.ts]: How about put these strings into the namespace?

File: src/Utils.ts [hunk 3]
```diff
@@ -1,29 +1,65 @@
...
+    export function getAiKey(): string {
+        return EXTENSION_AI_KEY;
+    }
+
+    export function getTempFolderPath(...args: string[]): string {
+        return path.join(os.tmpdir(), getExtensionId(), ...args);
+    }
+
+    export function getExtensionRootPath(...args: string[]): string {
+        return path.join(extensions.getExtension(getExtensionId()).extensionPath, ...args);
```
Comment (jdneo) [inline src/Utils.ts]: how about passing the context as an argument and call `context.extensionPath`

Comment (Eskibear) [reply src/Utils.ts]: `context` cannot be accessed elsewhere, I think it's better to get extension path by id

**Review by jdneo — APPROVED** [review]

**Pull request merged**